src/components/screens/monitor/delegates/withForgingStatus.js

```diff
@@ -188,6 +193,18 @@ const withForgingStatus = delegatesKey => (ChildComponent) => {
       }
     }
 
+    ensureGeneratorUsername(block) {
+      const generator = this.state.nextForgers.find(
```

Comment:
Now that `nextForgers` is a list instead of an object it would be nice for performance:
- 🐎 to avoid the `find` in case `generator` is not needed, that is if `block.generatorUsername` is defined.
- 🐎 prevent maping on all blocks below in this file if only first is used, something like replace:
```js
latestBlock: latestBlocks.map(this.ensureGeneratorUsername.bind(this))[0]
```
with
```js
latestBlock: latestBlocks.slice(0,1).map(this.ensureGeneratorUsername.bind(this))[0]
```